curl.nix

```diff
@@ -0,0 +1,12 @@
+{ mkDerivation, base, bytestring, containers, curlFull, stdenv, hpack }:
+mkDerivation {
+  pname = "curl";
+  version = "1.3.8";
+  src = ./.;
```

Comment:
This should filter out all files that are irrelevant to the build (like `.git`, `.stack-work`, ...). Do we need this file here at all, actually?

Reply:
We do need this path, otherwise it tries to build from non-local sources.

Reply:
I used `stdenv.lib.cleanSource` which should be enough for here.